classes/monograph/ChapterDAO.inc.php

```diff
@@ -109,7 +109,7 @@ public function getBySourceChapterId(int $sourceChapterId, bool $orderByPublicat
                 'SELECT	*
                 FROM submission_chapters
                 WHERE source_chapter_id = ?
-                OR chapter_id = ?'
+                OR ISNULL(source_chapter_id) AND chapter_id = ?'
```

Comment:
This caused a fatal error for me in `postgres`. I fixed it by changing this to `OR source_chapter_id IS NULL AND`. Also, do we need parentheses to force the the `OR` to work right?

```
WHERE A OR (B AND C)
```

Reply:
Okay, I change this. We don't need parentheses, because `AND` should always running before `OR`. You can see that here [https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-PRECEDENCE](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-PRECEDENCE) and here [https://dev.mysql.com/doc/refman/8.0/en/operator-precedence.html](https://dev.mysql.com/doc/refman/8.0/en/operator-precedence.html). However, I will add parentheses to make the query better readable.